src/shared/services/app-globals.service.ts

```diff
@@ -22,27 +22,86 @@
 
 import { Injectable, TemplateRef } from '@angular/core';
 import { ReplaySubject } from 'rxjs/ReplaySubject';
-import { SchemaValidationErrors } from '../interfaces';
+import { CategorizedValidationErrors, SchemaValidationErrors } from '../interfaces';
 
 @Injectable()
 export class AppGlobalsService {
-  private _globalErrorsSubject: ReplaySubject<SchemaValidationErrors> = new ReplaySubject<any>(1);
+  private _externalCategorizedErrorsSubject = new ReplaySubject<CategorizedValidationErrors>(1);
+  private _internalCategorizedErrorsSubject = new ReplaySubject<CategorizedValidationErrors>(1);
+  private _externalErrorCountersSubject = new ReplaySubject<{errors: number, warnings: number}>(1);
+  private _internalErrorCountersSubject = new ReplaySubject<{errors: number, warnings: number}>(1);
   public adminMode = false;
   public activeTabName = '';
   public tabNameToFirstTopLevelElement: { [tabName: string]: string } = {};
   public templates: { [templateName: string]: TemplateRef<any> };
+  public internalErrorMap = {};
+  public internalCategorizedErrorMap: CategorizedValidationErrors = { Errors: {}, Warnings: {} };
+  public externalCategorizedErrorMap: CategorizedValidationErrors = { Errors: {}, Warnings: {} };
 
-  set globalErrors(errors: SchemaValidationErrors) {
-    this._globalErrorsSubject.next(errors);
+  get externalCategorizedErrorsSubject(): ReplaySubject<CategorizedValidationErrors> {
+    return this._externalCategorizedErrorsSubject;
   }
 
-  get globalErrorsSubject(): ReplaySubject<SchemaValidationErrors> {
-    return this._globalErrorsSubject;
+  get externalErrorCountersSubject(): ReplaySubject<{errors: number, warnings: number}> {
+    return this._externalErrorCountersSubject;
+  }
+
+  get internalCategorizedErrorsSubject(): ReplaySubject<CategorizedValidationErrors> {
+    return this._internalCategorizedErrorsSubject;
+  }
+
+  get internalErrorCountersSubject(): ReplaySubject<{errors: number, warnings: number}> {
+    return this._internalErrorCountersSubject;
+  }
+
+  set externalErrors(errors: SchemaValidationErrors) {
+    this.externalCategorizedErrorMap = this.processErrorMap(errors);
+
+    this.externalCategorizedErrorsSubject.next(this.externalCategorizedErrorMap);
+    this.externalErrorCountersSubject.next({
+      errors: Object.keys(this.externalCategorizedErrorMap['Errors']).length,
```

Comment:
Since `externalCategorizedErrorMap` has for each key aggregated errors, this will not count all of them.